environment/src/main/java/com/sequenceiq/environment/environment/sync/EnvironmentSyncService.java

```diff
@@ -1,24 +1,103 @@
+import static com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status.STOPPED;
+import static com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status.STOP_FAILED;
+import static com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status.STOP_IN_PROGRESS;
+import static com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status.STOP_REQUESTED;
+import static com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status.UNHEALTHY;
+import static com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status.UNKNOWN;
+import static com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status.UNREACHABLE;
+import static com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status.UPDATE_FAILED;
+import static com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status.UPDATE_IN_PROGRESS;
+import static com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status.UPDATE_REQUESTED;
+import static com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status.UPGRADE_CCM_FAILED;
+import static com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status.UPGRADE_CCM_IN_PROGRESS;
+import static com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status.UPGRADE_CCM_REQUESTED;
+import static com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status.UPSCALE_FAILED;
+import static com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status.VERTICAL_SCALE_FAILED;
+import static com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status.WAIT_FOR_SYNC;
 
+import java.util.Map;
 import java.util.Optional;
 
 import org.springframework.stereotype.Component;
 
+import com.google.common.annotations.VisibleForTesting;
 import com.sequenceiq.environment.environment.EnvironmentStatus;
 import com.sequenceiq.environment.environment.domain.Environment;
 import com.sequenceiq.environment.environment.service.freeipa.FreeIpaService;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status;
 import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.describe.DescribeFreeIpaResponse;
 
 @Component
 public class EnvironmentSyncService {
 
+    private static final Map<Status, EnvironmentStatus> FREEIPA_STATUS_TO_ENV_STATUS_MAP = Map.ofEntries(
+            Map.entry(REQUESTED, FREEIPA_CREATION_IN_PROGRESS),
+            Map.entry(CREATE_IN_PROGRESS, FREEIPA_CREATION_IN_PROGRESS),
+            Map.entry(AVAILABLE, EnvironmentStatus.AVAILABLE),
+            Map.entry(STACK_AVAILABLE, FREEIPA_CREATION_IN_PROGRESS),
+            Map.entry(DIAGNOSTICS_COLLECTION_IN_PROGRESS, EnvironmentStatus.AVAILABLE),
+            Map.entry(UPDATE_IN_PROGRESS, UPDATE_INITIATED),
+            Map.entry(UPDATE_REQUESTED, UPDATE_INITIATED),
+            Map.entry(UPDATE_FAILED, EnvironmentStatus.UPDATE_FAILED),
+            Map.entry(UPSCALE_FAILED, EnvironmentStatus.AVAILABLE),
+            Map.entry(DOWNSCALE_FAILED, EnvironmentStatus.AVAILABLE),
```

Comment:
Is the FreeIPA indeed healthy in these cases?

Reply:
There is still at least 1 instance of FreeIPA operating.